### DISCUSSION TITLE
User Submitted Shaders

### DISCUSSION THREAD

**Discussion by VegaDeftwing** [noisedeck/noisedeck-docs#28] [Ideas]
The Live-Code module is cool, but I think adding a community element to it would be nice. Maybe a public repo could be setup (I know noisedeck is a business- so a CLA is understandable) for adding live-code presets? I'm not sure how you'd avoid making the list super long, maybe 'folders' in the drop down to select by user or category?

also, as a 2-for-1 on this feature request, custom post processing and mixing shaders could be neat too. I don't know if the live-code module's code is easily reused for that though.

**Comment by jesshewitt** [comment]
Alex is working on a design for user modules, so we are definitely on the same page there! 

Live-code versions of mixer and post modules would be great! Shouldn't be too hard to implement.

**Reply by VegaDeftwing** [reply]
Sweet! I think the way VCV Rack has done their module store is pretty clean, might be an avenue worth exploring. They let users either submit open source modules to the library or sell them at a fair profit cut. That might incentivize further development of more complex shaders by the community
